exonum-java-binding/core/rust/src/utils/resource_manager/imp.rs

```diff
@@ -160,111 +189,129 @@ mod tests {
     use std::i64;
 
     enum T {}
+    enum HandleFooType {}
+    enum HandleBarType {}
+    const HANDLE_FOO: Handle = 1000;
+    const HANDLE_BAR: Handle = 2000;
     const INVALID_HANDLE: Handle = i64::MAX;
-
-    // Unique ("valid") handles should be used in the each test because `HANDLES_MAP` is a shared
-    // state and tests are run concurrently.
-    const MANAGE_HANDLES_FIRST_HANDLE: Handle = 1000;
-    const MANAGE_HANDLES_SECOND_HANDLE: Handle = 2000;
-    const MANAGE_HANDLES_NON_OWNED_HANDLE: Handle = 3000;
-    const DUPLICATED_HANDLE: Handle = 4000;
-    const WRONG_TYPE_HANDLE: Handle = 5000;
-    const WRONG_OWNERSHIP_HANDLE: Handle = 6000;
+    const ZERO_HANDLE: Handle = 0;
 
     #[test]
-    fn manage_handles() {
-        // Add Java-owned handle.
-        enum T1 {}
-        add_handle::<T1>(MANAGE_HANDLES_FIRST_HANDLE);
-        check_handle::<T1>(MANAGE_HANDLES_FIRST_HANDLE);
-        check_handle_impl::<T1>(
-            MANAGE_HANDLES_FIRST_HANDLE,
-            Some(HandleOwnershipType::JavaOwned),
+    fn check_handle_type_and_ownership_ok() {
+        let handle_info = prepare_handle_info();
```

Comment:
Here and elsewhere: I do not think `prepare_handle_info` works in such cases because subsequent assertions *do* rely on what it puts there. Either
* This method must be parameterized.
* Converted to a builder, relevant to the test data must be set up in each test that depends on that data
* If no parameterization needed — given a name that reflects what it creates.

Currently it is not clear what all these tests using `prepare_handle_info` test.

Reply:
Please see also https://testing.googleblog.com/2018/02/testing-on-toilet-cleanly-create-test.html